Unable to build on latest rust nightly

Unable to build on latest Rust nightly with latest version. Rust compiler is suggesting to add `#![feature(const_fn_trait_bound)]`.

## Versions
* rustc 1.54.0-nightly (4de757209 2021-05-01)
* cargo 1.53.0-nightly (f3e13226d 2021-04-30)

## Stack

```shell
rust_1      | error[E0658]: trait bounds other than `Sized` on const fn parameters are unstable
rust_1      |    --> /usr/local/cargo/git/checkouts/ledger-nanos-sdk-f77eb1a6021b7ee5/56c5859/src/nvm.rs:181:6
rust_1      |     |
rust_1      | 181 | impl<T> AtomicStorage<T> where T: Copy {
rust_1      |     |      ^
rust_1      |     |
rust_1      |     = note: see issue #57563 <https://github.com/rust-lang/rust/issues/57563> for more information
rust_1      |     = help: add `#![feature(const_fn_trait_bound)]` to the crate attributes to enable
rust_1      |
rust_1      | error[E0658]: trait bounds other than `Sized` on const fn parameters are unstable
rust_1      |    --> /usr/local/cargo/git/checkouts/ledger-nanos-sdk-f77eb1a6021b7ee5/56c5859/src/nvm.rs:237:6
rust_1      |     |
rust_1      | 237 | impl<T, const N: usize> Collection<T, N> where T: Copy {
rust_1      |     |      ^
rust_1      |     |
rust_1      |     = note: see issue #57563 <https://github.com/rust-lang/rust/issues/57563> for more information
rust_1      |     = help: add `#![feature(const_fn_trait_bound)]` to the crate attributes to enable
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,3 +23,4 @@ lto = true
 
 [features]
 speculos = []
+pre1_54 = []

--- a/README.md
+++ b/README.md
@@ -28,6 +28,12 @@ required.
 - install [Clang](http://releases.llvm.org/download.html).
 - install an [ARM GCC toolchain](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm/downloads)
 
+### Building with rustc < 1.54
+
+Building before rustc 1.54 should fail with `error[E0635]: unknown feature const_fn_trait_bound`.
+
+This is solved by activating a specific feature: `cargo build --features pre1_54`
+
 ## Contributing
 
 Make sure you've followed the installation steps above. In order for your PR to be accepted, it will have to pass the CI, which performs the following checks:

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -6,6 +6,7 @@
 #![feature(const_fn)]
 #![feature(asm)]
 #![feature(const_panic)]
+#![cfg_attr(not(feature = "pre1_54"), feature(const_fn_trait_bound))]
 
 pub mod bindings;
 pub mod buttons;

